lib/stack_master/sparkle_formation/compile_time/number_validator.rb

```diff
@@ -0,0 +1,35 @@
+require_relative 'value_validator'
+
+module StackMaster
+  module SparkleFormation
+    module CompileTime
+      class NumberValidator < ValueValidator
+
+        def initialize(name, definition, parameter)
+          @name = name
+          @definition = definition
+          @parameter = parameter
+        end
+
+        private
+
+        def check_is_valid
+          return true unless @definition[:type] == :number
+          invalid_values.empty?
+        end
+
+        def invalid_values
+          values = build_values(@definition, @parameter)
+          #TODO: check value is a number
+          values.reject {|value|
+            value.is_a?(Numeric) || value.is_a?(String) && value.to_s =~ /^(\d+)(\.?(\d+))?$/}
```

Comment:
From the Github Ruby style guide:
> Prefer {...} over do...end for single-line blocks. Avoid using {...} for multi-line blocks 

See https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md

Reply:
Further I find the logic here quite confusing... multiple and/or logic in a single line combined with regular expressions makes it hard to understand. Can this be broken down more?